cmd/nerdctl/run_log_driver_syslog_test.go

```diff
@@ -0,0 +1,509 @@
+	}
+	tml := x509.Certificate{
+		// you can add any attr that you need
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(5, 0, 0),
+		// you have to generate a different serial number each execution
+		SerialNumber: big.NewInt(123123),
+		Subject: pkix.Name{
+			CommonName:   "New Name",
+			Organization: []string{"New Org."},
+		},
+		BasicConstraintsValid: true,
+		IPAddresses:           []net.IP{net.ParseIP("127.0.0.1")},
+	}
+	ca, err := x509.CreateCertificate(rand.Reader, &tml, &tml, &key.PublicKey, key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	caOut, err := os.Create(caPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer caOut.Close()
+	// Generate a pem block with the certificate
+	if err := pem.Encode(caOut, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: ca,
+	}); err != nil {
+		return nil, nil, err
+	}
+
+	certTml := x509.Certificate{
+		// you can add any attr that you need
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(5, 0, 0),
+		// you have to generate a different serial number each execution
+		SerialNumber: big.NewInt(123123),
+		Subject: pkix.Name{
+			CommonName:   "New Name",
+			Organization: []string{"New Org."},
+		},
+		BasicConstraintsValid: true,
+		IPAddresses:           []net.IP{net.ParseIP("127.0.0.1"), net.IPv6loopback},
+	}
+	cert, err := x509.CreateCertificate(rand.Reader, &certTml, &tml, &key.PublicKey, key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	certOut, err := os.Create(certPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer caOut.Close()
+	// Generate a pem block with the certificate
+	if err := pem.Encode(certOut, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert,
+	}); err != nil {
+		return nil, nil, err
+	}
+
+	return &certificate{CA: caPath, Key: keyPath, Cert: certPath}, func() error {
+		if err := os.Remove(keyPath); err != nil {
+			return err
+		}
+		if err := os.Remove(certPath); err != nil {
+			return err
+		}
+		return os.Remove(caPath)
+	}, nil
+}
+
+func runPacketSyslog(c net.PacketConn, done chan<- string) {
+	var buf [4096]byte
+	var rcvd string
+	ct := 0
+	for {
+		var n int
+		var err error
+
+		_ = c.SetReadDeadline(time.Now().Add(100 * time.Millisecond))
+		n, _, err = c.ReadFrom(buf[:])
+		rcvd += string(buf[:n])
+		if err != nil {
+			if oe, ok := err.(*net.OpError); ok {
+				if ct < 3 && oe.Temporary() {
+					ct++
+					continue
+				}
+			}
+			break
+		}
+	}
+	c.Close()
+	done <- rcvd
+}
+
+func runStreamSyslog(l net.Listener, done chan<- string) {
+	for {
+		var c net.Conn
+		var err error
+		if c, err = l.Accept(); err != nil {
+			return
+		}
+		go func(c net.Conn) {
+			_ = c.SetReadDeadline(time.Now().Add(5 * time.Second))
+			b := bufio.NewReader(c)
+			for ct := 1; ct&7 != 0; ct++ {
+				s, err := b.ReadString('\n')
+				if err != nil {
+					break
+				}
+				done <- s
+			}
+			c.Close()
+		}(c)
+	}
+}
+
+func startServer(n, la string, done chan<- string, certs ...*certificate) (addr string, sock io.Closer) {
```

Comment:
```suggestion
func startSyslogServer(n, la string, done chan<- string, certs ...*certificate) (addr string, sock io.Closer) {
```

Reply:
Or maybe move to `pkg/testutil/syslogtest`

Reply:
Done with moved